### ISSUE TITLE
🐛 Bug Report: Auth backend has dependencies to modules

### ISSUE THREAD

**Issue by drodil** [backstage/backstage#24256]
### 📜 Description

Auth backend has dependencies to a lot of auth modules. These probably should not be here but instead only the modules needed should be added by the app.

### 👍 Expected behavior

No extra dependencies in the auth module

### 👎 Actual Behavior with Screenshots

https://github.com/backstage/backstage/blob/master/plugins/auth-backend/package.json#L46C1-L56C73

### 👟 Reproduction steps

See: https://github.com/backstage/backstage/blob/master/plugins/auth-backend/package.json#L46C1-L56C73

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by pjungermann** [comment]
The previous iterations got extracted and reexported by the auth-backend. To break these dependencies, a breaking change is required (as this is still a 0.x package, a minor version bump).

Also, not all auth providers got extracted and provided as modules, yet.

**Comment by freben** [comment]
Yeah this is unfortunately there because of the `providers` export in its API, which is what people who have not yet migrated to the new systems use.

When we make the breaking change to no longer support the old system, we'll definitely kill these dependencies.

**Comment by drodil** [comment]
Not stale
